### PR TITLE
@/charterafrica/components/GranteeCard #322

### LINKED ISSUE #322: @/charterafrica/components/GranteeCard
### Description

Reusing code (or knowledge) from Spotlight and Post card components, implement the `GranteeCard` component to be used in about [page](https://www.figma.com/file/gbnaO1FYPq16V2gAFbLOkl/CfA---EPD-Charter-Africa---Sep-2022?node-id=9108%3A28547&t=gHql57tAK6HPaLGe-0).

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #322 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Screenshots
image

image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/charterafrica/src/pages/[slug]/[slug].spec.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const { test, expect } = require("@playwright/test");
```
Comment (kilemensi) [inline apps/charterafrica/src/pages/[slug]/[slug].spec.js:1]: Very good thinking... as you can see my WIP in #333 , we can take this further and actually delete all pages, leaving just `[...slug].page.js` and `index.page.js`.

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: apps/charterafrica/src/lib/data/common/index.js [hunk 2]
```diff
@@ -100,8 +100,52 @@ export async function processPageResearch({ blocks }) {
...
+        alt: `Research ${i}`,
+        prefix: "media",
+        filename: "Rectangle 113.jpg",
+        mimeType: "image/jpg",
+        filesize: 257010,
+        width: 1236,
+        height: 696,
+        createdAt: "2023-01-26T11:21:14.868Z",
+        updatedAt: "2023-01-26T11:21:14.868Z",
+        url: "http://localhost:3000/media/Rectangle 117.png",
```
Comment (kelvinkipruto) [inline apps/charterafrica/src/lib/data/common/index.js:130]: I think it should refer to this [image](https://github.com/CodeForAfrica/ui/blob/7e9d370eef9c39f650bd649d5a59403c6eef3fa3/apps/charterafrica/public/images/the-charter-project-africa_brandguide_guidedemarque-2-2.jpg)
Also use relative path i.e `/images/....` incase someone runs it using a different port

File: apps/charterafrica/src/components/StyledCard/StyledCard.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Card, styled } from "@mui/material";
+
+import { neutral } from "@/charterafrica/colors";
+
+const StyledCard = styled(Card)(({ ownerState, theme }) => ({
+  backgroundColor: "#fff",
+  transition: theme.transitions.create(["filter", "border-color"]),
+  width: "100%",
+  "&:hover": {
```
Comment (kelvinkipruto) [inline apps/charterafrica/src/components/StyledCard/StyledCard.js:9]: From the designs, it looks like when hovering over the image, there should also be a similar hover effect

File: apps/charterafrica/src/components/Grantees/GranteeCard.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            size="small"
+            variant="contained"
+            color="success"
+            href={primaryLink.href}
+          >
+            {primaryLink.label}
+          </Button>
+          <br />
+          <Button
+            component={Link}
```
[the remark this answers is not in the document]
Comment (koechkevin) [reply apps/charterafrica/src/components/Grantees/GranteeCard.js:92]: I left a comment on figma designs

File: apps/charterafrica/src/components/StyledCard/StyledCard.js (hunk 3 above)
Comment (koechkevin) [reply apps/charterafrica/src/components/StyledCard/StyledCard.js:9]: @kelvinkipruto not sure I get this

Comment (kelvinkipruto) [reply apps/charterafrica/src/components/StyledCard/StyledCard.js:9]: @koechkevin When hovering over the `CardMedia`, on `GuaranteeCard` there is a border that appears

File: apps/charterafrica/src/components/Grantees/GranteeCard.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  return (
+    <Card
+      elevation={elevation}
+      sx={sx}
+      ref={ref}
+      ownerState={ownerState}
+      variant={variant}
+    >
+      <StyledActionArea component={href ? Link : undefined} href={href}>
+        <CardMedia image={image.url} sx={{ height: 200 }} />
```
Comment (kelvinkipruto) [inline apps/charterafrica/src/components/Grantees/GranteeCard.js:52]: Shouldn't the height be 264?

**Comment by kilemensi** [comment]
Any progress here @koechkevin ? Need this to continue with other about page components in #333

File: apps/charterafrica/src/components/StyledCard/StyledCard.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      backgroundColor: "#fff",
+    }),
+  },
+  ...(ownerState.variant === "outlined" && {
+    border: `2px solid transparent`,
+    borderRadius: 5,
+    filter: "drop-shadow(0px 6px 12px rgba(0, 0, 0, 0.1))",
+    "& .MuiCardActionArea": {
+      border: `2px solid transparent`,
+    },
```
Comment (kilemensi) [inline apps/charterafrica/src/components/StyledCard/StyledCard.js:22]: Can't these be combined?
```jsx
  ...(ownerState.variant === "outlined" && {
    border: `2px solid transparent`,
    ... other normal state styles,
    "&:hover": {
      borderColor: neutral[700],
      ... other hover styles,
    }.
  }

File: apps/charterafrica/src/pages/[...slugs].page.js [hunk 7]
```diff
@@ -17,6 +18,14 @@ function Opportunities({ blocks }) {
...
       case "page-info":
         return <PageInfo {...block} key={block.slug} />;
+      case "grantees":
+        return <Grantees {...block} key={block.slug} />;
+      case "explainers":
+        return <Explainers {...block} key={block.slug} />;
+      case "news":
+        return <Articles {...block} key={block.slug} />;
+      case "research":
+        return <Articles {...block} key={block.slug} />;
```
Comment (kilemensi) [inline apps/charterafrica/src/pages/[...slugs].page.js:28]: Order these alphabetically.

File: apps/charterafrica/src/pages/index.spec.js [hunk 8]
```diff
@@ -5,3 +5,9 @@ test("/", async ({ page }) => {
   const title = await page.title();
   expect(title).toBe("charter.AFRICA");
 });
+
+test("/about", async ({ page }) => {
+  await page.goto("/about");
+  const title = await page.title();
+  expect(title).toBe("About");
+});
```
Comment (kilemensi) [inline apps/charterafrica/src/pages/index.spec.js:13]: Move this into `[...slug].page.spec.js`

**Review by kilemensi — APPROVED** [review]
🚀

**Review by kelvinkipruto — APPROVED** [review]

**Pull request merged**